src/tabular-api-surface.ts

```diff
@@ -224,6 +224,19 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
 
     rowKeys = options.keys || [];
 
+    /*
+    The following line of code sets the timeout if it was provided while
+    creating the client. This will be used to determine if the client should
+    retry on DEADLINE_EXCEEDED errors. Eventually, this will be handled
+    downstream in google-gax.
+     */
+    const timeout =
```

Comment:
I think timeout should come from [getRowsOptions](https://github.com/googleapis/nodejs-bigtable/blob/main/src/tabular-api-surface.ts#L90) gaxOptions -> timeout https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html#timeout

Reply:
In this conformance test the timeout is provided in the create-client test proxy service, not the readRows test proxy service. So for the conformance test to pass we have to read the timeout from the client. See the screenshots below for proof of this.

![image](https://github.com/user-attachments/assets/cf28460e-5e52-41a8-a3aa-84b63c4265c4)
![image](https://github.com/user-attachments/assets/544a23d2-5b0f-4c91-a809-6320ae9ebb61)

That said, I understand that if a timeout is provided in `GetRowOptions` then it should be used instead of the timeout set in the client so I can change that.

Reply:
We're not just trying to make the test pass, we're trying to verify that the client behaves as expected. Will a customer be able to configure the overall timeout on the client directly? If not, we should always use the timeout from `GetRowsOptions` as the source of truth, and figure out how we can fix the test proxy so the client opts from conformance test is passed in correctly.

Reply:
> Will a customer be able to configure the overall timeout on the client directly?

In the way that I had it before, yes. They do it by setting the clientConfig when creating the client [like so](https://github.com/googleapis/nodejs-bigtable/pull/1533/files#diff-167461f4f7152422b73f5542871adc43acb88b564b420fde31736a3442168ef0L62). That said, I have made the change so that the timeout can be overridden for each call as you suggested.

Note right now, that the timeout is not being passed into the readRows test proxy service at all. So the test runner needs to change to pass the timeout along in the readRows call if you want to see that this timeout is being is used from `GetRowOptions`.

Reply:
Note that the test will still pass because it looks for `timeout` passed into the method and if it is `undefined` it will use the timeout provided when the client was created which happens to be 2 seconds for this test. So the test runner doesn't need to change now for this test, but maybe a test for that `timeout` parameter should be added in the future to the test runner.

```
      opts?.gaxOptions?.timeout ||
      (this?.bigtable?.options?.BigtableClient?.clientConfig?.interfaces &&
        this?.bigtable?.options?.BigtableClient?.clientConfig?.interfaces[
          'google.bigtable.v2.Bigtable'
        ]?.methods['ReadRows']?.timeout_millis);
```

Reply:
I see that the gaxOption is hard to test with the current conformance test set up. Can we create a unit test to test this timeout is used?

Reply:
Sure. Mocking errors and responses from a stream is somewhat cumbersome so I'll just need a bit of time to get that set up.